src/mono/browser/runtime/strings.ts

```diff
@@ -244,6 +244,7 @@ function storeStringInInternTable(string: string, root: WasmRoot<MonoString>, in
 
 function stringToMonoStringNewRoot(string: string, result: WasmRoot<MonoString>): void {
     const bufferLen = (string.length + 1) * 2;
+    // TODO this could be stack allocated
```

Comment:
strings are unbounded, so there wouldn't necessarily be space on the stack. it would be a good optimization for sufficiently small strings though.